docs/index.md

```diff
@@ -84,26 +84,175 @@ connection "terraform" {
+**Note:** If paths include `*`, all files (including non-Terraform configuration files) in the CWD will be matched, which may cause errors if incompatible file types exist.
+
+#### Configuring remote Git repositories URLs
+
+Not only local files, but you can also configure `paths` with any remote repository URLs (i.e. GitHub, BitBucket, GitLab) to search various Terraform configuration files in it.
+
+You can also mention [wildcards](https://pkg.go.dev/path/filepath#Match) and support `**` for recursive matching. For example:
+
+- `github.com/turbot/steampipe-plugin-aws//*.tf` matches all top-level Terraform configuration files in the specified repository.
+- `github.com/turbot/steampipe-plugin-aws//**/*tf` matches all Terraform configuration files in the specified repository and all sub-directories.
+- `github.com/turbot/steampipe-plugin-aws//**/*tf?ref=fix_7677` matches all Terraform configuration files in the specific tag of a repository.
+
+You can specify a subdirectory after a double-slash (`//`) if you want to download only a specific subdirectory from a downloaded directory.
+
+```hcl
+connection "terraform" {
+  plugin = "terraform"
+
+  paths = [ "github.com/turbot/steampipe-plugin-aws//aws-test/tests/aws_acm_certificate//*.tf" ]
+}
+```
+
+Similarly, you can define a list of GitLab and BitBucket URLs to search for the Terraform configuration files.
+
+```hcl
+connection "terraform" {
+  plugin = "terraform"
+
+  paths = [
+    "github.com/turbot/steampipe-plugin-aws//**/*tf",
+    "github.com/hashicorp/terraform-guides//infrastructure-as-code//**/*.tf",
+    "bitbucket.org/benturrell/terraform-arcgis-portal//modules/shared//*.tf",
+    "bitbucket.org/benturrell/terraform-arcgis-portal//modules//**/*.tf",
+    "gitlab.com/gitlab-org/configure/examples/gitlab-terraform-aws//*.tf",
+    "gitlab.com/gitlab-org/configure/examples/gitlab-terraform-aws//**/*.tf"
+  ]
+}
+```
+
+#### Configuring S3 URLs
+
+As a part of reading files from remote, you can also query all Terraform configuration files stored inside an S3 bucket (public or private) using the URL. For example:
+
+- **Accessing a private bucket**
+
+  In order to access your files in a private S3 bucket, you will need to configure your credentials. You can use your configured AWS profile from local `~/.aws/config`, or pass the credentials using the standard AWS environment variables (i.e. `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_REGION`).
+
+  **Note:** Make sure that `region` is configured in the config. If not set in the config, `region` will be fetched from the standard environment variable (i.e. `AWS_REGION`).
+
+  You can authenticate your request by using the above credential option in the URL path. For example:
+
+  ```hcl
+  connection "terraform" {
+    plugin = "terraform"
+
+    paths = [
+      "s3::https://bucket-2.s3.us-east-1.amazonaws.com//*.tf?aws_profile=<AWS_PROFILE>",
```

Comment:
@cbruno10, if `region` is not set using the environment variable, it will check for the `region` value set in the config. If it is not set there, the request will fail with a `MissingRegion` error.